src/components/paragraph/rows/three-column.tsx

```diff
@@ -1,32 +1,46 @@
-import Paragraph, {getParagraphBehaviors} from "@/components/paragraph"
+import OneColumn from "@/components/paragraph/rows/one-column"
 import {ParagraphUnion} from "@/lib/gql/__generated__/drupal.d"
+import {getParagraphBehaviors} from "@/components/paragraph"
+import {isPreviewMode} from "@/lib/drupal/is-draft-mode"
 
 interface LayoutProps {
   items: ParagraphUnion[]
   fullWidth?: boolean
+  config?: Record<string, string>
 }
 
-const ThreeColumn = ({items}: LayoutProps) => {
-  const leftItems = items.filter(item => getParagraphBehaviors(item)?.layout_paragraphs?.region === "left")
-  const mainItems = items.filter(item => getParagraphBehaviors(item)?.layout_paragraphs?.region === "main")
-  const rightItems = items.filter(item => getParagraphBehaviors(item)?.layout_paragraphs?.region === "right")
+const ThreeColumn = ({items, fullWidth = true}: LayoutProps) => {
+  const leftItems = items.filter(item => getParagraphBehaviors(item).layout_paragraphs?.region === "left")
+  const mainItems = items.filter(
+    item => !["left", "right"].includes(getParagraphBehaviors(item).layout_paragraphs?.region || "main")
+  )
+  const rightItems = items.filter(item => getParagraphBehaviors(item).layout_paragraphs?.region === "right")
+
+  const draftProps: Record<string, string> = {}
+  if (isPreviewMode()) {
+    draftProps["data-columns"] = "3"
+  }
+
   return (
-    <div data-rows="three-column" className="centered grid gap-[90px] lg:grid-cols-3">
-      <div className="gap-2xll relative flex min-w-0 flex-col">
-        {leftItems.map(item => (
-          <Paragraph key={item.id} paragraph={item} fullWidth={false} />
-        ))}
-      </div>
-      <div className="relative flex min-w-0 flex-col gap-2xl">
-        {mainItems.map(item => (
-          <Paragraph key={item.id} paragraph={item} fullWidth={false} />
-        ))}
-      </div>
-      <div className="relative flex min-w-0 flex-col gap-2xl">
-        {rightItems.map(item => (
-          <Paragraph key={item.id} paragraph={item} fullWidth={false} />
-        ))}
-      </div>
+    <div
+      className="centered flex flex-col justify-between gap-[9rem] md:flex-row md:flex-wrap xl:flex-nowrap"
+      {...draftProps}
+    >
+      <OneColumn
+        className="w-full md:w-[calc(50%_-_5rem)] xl:w-[calc(33.3%_-_5rem)]"
+        items={leftItems}
+        fullWidth={fullWidth}
+      />
+      <OneColumn
+        className="w-full md:w-[calc(50%_-_5rem)] xl:w-[calc(33.3%_-_5rem)]"
+        items={mainItems}
+        fullWidth={fullWidth}
+      />
+      <OneColumn
+        className="mx-auto w-full md:w-[calc(50%_-_5rem)] xl:w-[calc(33.3%_-_5rem)]"
+        items={rightItems}
+        fullWidth={fullWidth}
```

Comment:
@pookmish 🤔 am I overthinking this?

Reply:
ideally you wouldn't even need the classes on the one column. but that's pretty much it.

Reply:
Should we move this portion to another PR?

Reply:
@pookmish If you’re anticipating a more extensive review and refinement, I’d be happy to move it to another PR! The three-column layout with the 2/1 stacking was a specific request from Darcy as part of the homepage redesign so we could also keep it together as well 😃 
![image](https://github.com/user-attachments/assets/44bd0b65-674b-4f06-a7f3-cf99e7275578)